Added workflow to test kafka, rabbitmq and sqs connector

girishg4t opened the pull request:
Added workflow for kafka, rabbitmq, sqs.
This workflow does the following steps
1) Create the docker image of the connector
2) Create the respective queue
3) Create and deploy the http server
4) Deploy the connector and queue
5) Verify the connector by sending messages in the queue

vishal-biyani reviewed (changes requested):
Hi @girishg4t Thanks for this. I have left some comments inside the PR but also some generic comments:

1) Do we need integration/rabbitmq/publisher/.publish-msgs.go.swn checked in?

2) Thinking about the overall structure of tests - I am thinking that we should have one test directory at repo level - which has all common things such as HTTP server, it's Dockerfile, deployment manifest, etc. Then in each connector folder - we have a test folder which has test files related to that connector only.

sahil-lakhwani commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,285 @@
+    if: ${{ github.event.label.name == 'rabbitmq' }} || ${{ github.event.label.name == 'Rabbitmq' }}${{ github.event.label.name == 'RABBITMQ' }}
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout the current repo
+        uses: actions/checkout@v1
+      - uses: dorny/paths-filter@v2
+        id: filter
+        with:
+          filters: |
+            rabbitmqchanges:
+              - 'integration/rabbitmq/**'    
+              - 'rabbitmq-http-connector/**'                   
+      - name: Install Kind Cluster
+        if: steps.filter.outputs.rabbitmqchanges == 'true'      
+        run: |
+          curl -Lo ./kind $KIND_URL/$KIND_VER/$KIND_ENV
+          chmod +x ./kind
+          sudo mv ./kind /usr/bin/
+        env:
+          KIND_URL: https://kind.sigs.k8s.io/dl
+          KIND_VER: v0.8.1
+          KIND_ENV: kind-linux-amd64
+      - name: Install KEDA using HELM
+        if: steps.filter.outputs.rabbitmqchanges == 'true'        
+        run: |
+          cd integration/
+          ./registry.sh
+          helm repo add kedacore https://kedacore.github.io/charts
+          helm repo update
+          kubectl create namespace keda
+          helm install keda kedacore/keda --namespace keda
```
So different jobs have different contexts, hence I am using a single job with different steps which you can see here: https://github.com/sahil-lakhwani/environments/blob/master/.github/workflows/environment.yaml

girishg4t replied:
I have corrected the flow and now i am using single job with filters which will trigger only if there are changes in respective files and moved the common stuffs at top

girishg4t commented:
> Hi @girishg4t Thanks for this. I have left some comments inside the PR but also some generic comments:
> 
>     1. Do we need integration/rabbitmq/publisher/.publish-msgs.go.swn checked in?
> 
>     2. Thinking about the overall structure of tests - I am thinking that we should have one test directory at repo level - which has all common things such as HTTP server, it's Dockerfile, deployment manifest, etc. Then in each connector folder - we have a test folder which has test files related to that connector only.

Removed the file mentioned in point 1
Changed the folder structure as per review comments

girishg4t commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,285 @@
+    branches:
+      - master
+  pull_request:
+    types: [ labeled ]
```
There was condition in each job which was checking for label, now removed that because we are using filters

girishg4t commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,285 @@
+          filters: |
+            kafkachanges:
+              - 'integration/kafka/**'    
+              - 'kafka-http-connector/**'
```
yes removed that

girishg4t commented on .github/workflows/ci.yml:
```diff
@@ -0,0 +1,285 @@
+              - 'integration/kafka/**'    
+              - 'kafka-http-connector/**'
+      - name: Install Kind Cluster
+        if: steps.filter.outputs.kafkachanges == 'true'
```
moved cluster creating and other common thing at single place

girishg4t commented on aws-sqs-http-connector/workflow/kubernetes/Dockerfile:
```diff
@@ -0,0 +1,12 @@
+
```
We are using localstack for testing sqs connector and to connect to localstack from our deployment we require awscli, since our sqs connector images does not contain awscli so i have created a separate image with all the required things to connect to localstack.

There can be different ways of doing this, i followed this approach

girishg4t commented on aws-sqs-http-connector/workflow/kubernetes/localstack.yaml:
```diff
@@ -0,0 +1,43 @@
+  name: localstack
+spec:
+  # using the selector, we will expose the running deployments
+  # this is how Kubernetes knows, that a given service belongs to a deployment
```
removed

girishg4t commented on aws-sqs-http-connector/workflow/kubernetes/localstack.yaml:
```diff
@@ -0,0 +1,43 @@
+  name: localstack
+spec:
+  # selector tells Kubernetes what Deployment this Service
+  # belongs to
```
removed

girishg4t commented on aws-sqs-http-connector/workflow/server/Dockerfile:
```diff
@@ -0,0 +1,6 @@
+FROM golang:1.12-alpine
```
same comment
https://github.com/fission/keda-connectors/pull/34#discussion_r500132720

girishg4t commented on aws-sqs-http-connector/workflow/server/main.go:
```diff
@@ -0,0 +1,64 @@
+package main
```
Workflow folder is not required removed that

vishal-biyani commented:
Thanks a ton for this one @girishg4t
